src/ace/ari_text/util.py

```diff
@@ -89,7 +89,7 @@ def t_decfrac(found):
 
 
 # float either contains a decimal point or exponent or both
-@TypeMatch.apply(r'[+-]?((\d+|\d*\.\d*)([eE][+-]?\d+)|\d*\.\d*|Infinity)|NaN')
+@TypeMatch.apply(r'(?i)[+-]?((\d+|\d*\.\d*)([eE][+-]?\d+)|\d*\.\d*|Infinity)|NaN')
```

Comment:
This came up as an error in python 3.12, which must be more strict than earlier ones.